packages/git/src/common/git.ts

```diff
@@ -691,12 +691,20 @@ export interface Git extends Disposable {
     show(repository: Repository, uri: string, options?: Git.Options.Show): Promise<string>;
 
     /**
-     * It resolves to an array of configured remotes for the given repository.
+     * It resolves to an array of configured remotes names for the given repository.
      *
-     * @param repository the repository to get the remotes.
+     * @param repository the repository to get the remote names.
      */
     remote(repository: Repository): Promise<string[]>;
 
+    /**
+     * It resolves to an array of configured remote objects for the given Git action.
+     *
+     * @param repository the repository to get the remote objects.
+     * @param type the git action either `fetch` or `push`.
+     */
+    remote(repository: Repository, options: { action: RemoteAction }): Promise<Remote[]>;
```

Comment:
Is it possible to do something like this?
```diff
diff --git a/packages/git/src/common/git.ts b/packages/git/src/common/git.ts
index 1dc87965..a5040c34 100644
--- a/packages/git/src/common/git.ts
+++ b/packages/git/src/common/git.ts
@@ -275,6 +275,18 @@ export namespace Git {

         }

+        /**
+         * Options for `git remote`.
+         */
+        export interface Remote {
+
+            /**
+             * Be a little more verbose and show remote URL after name.
+             */
+            readonly verbose?: true;
+
+        }
+
         /**
          * Options for further refining the `git show` command.
          */
@@ -696,6 +708,9 @@ export interface Git extends Disposable {
      * @param repository the repository to get the remotes.
      */
     remote(repository: Repository): Promise<string[]>;
+    // tslint:disable-next-line:no-any
+    remote(repository: Repository, options?: Git.Options.Remote): Promise<any[]>;
+    remote(repository: Repository, options: { verbose: true }): Promise<Remote[]>;

     /**
      * Executes the Git command and resolves to the result. If an executed Git command exits with a code that is not in the `successExitCodes` or an error not in `expectedErrors`,
```

We could omit the one with the `any` return type.